apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LogEcsReformatting.java

```diff
@@ -0,0 +1,33 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2021 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.logging;
+
+public enum LogEcsReformatting {
+    OFF,
+    SHADE,
+    REPLACE,
+    // NOT IMPLEMENTED YET
+    //OVERRIDE
```

Comment:
As I find it hard to recall what the difference between `REPLACE ` and `OVERRIDE ` is, just some ideas how we could maybe make it clearer.

```suggestion
    SHADE_FILE,
    REPLACE_FILE,
    // NOT IMPLEMENTED YET
    // REFORMAT
```

The `_FILE` options only work for file appenders while `REFORMAT` works with potentially any appender.

Reply:
I actually prefer the simplicity in enum names, especially ones you use for configuration, so I think these single-verb-word names make sense. I see what you say about the meaning of each, but I don't think this will make a big change - in any case you would read the documentation and this is where I put efforts to very clearly explain what each does.
In addition, `shade` is currently only applied for files because this is how I implemented it, but it may do the same for console/stdout appenders as well if required for the same reason (keep existing log stream but also ingest ECS format of it to ES).